ci/run_black.ps1

```diff
@@ -0,0 +1,19 @@
+$ExitCode = 0
+$GitDiff = git diff origin/master
+# If there is no diff between master, then black everything.
+if ( $GitDiff -eq $null ) {
+    black .
+    if ($LastExitCode -ne 0) { $ExitCode = $LastExitCode }
+}
+# Else black just the diff.
+else {
+    Write-Output $GitDiff | black
+    if ($LastExitCode -ne 0) { $ExitCode = $LastExitCode }
+    # If the project_template changed, then black the testproject too.
+    $GitDiffTempl = Write-Output $GitDiff | Select-String -Pattern "^diff .*/project_template/.*"
+    if ( $GitDiffTempl -ne $null ) {
+        black .
```

Comment:
The purpose of this statement is to run black on the testproject IF any code in the project_template changed. So here it should only be checking `testproject/`, not the entire directory.